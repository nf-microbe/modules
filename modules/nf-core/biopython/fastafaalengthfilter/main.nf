process BIOPYTHON_FASTAFAALENGTHFILTER {
    tag "${meta.id}"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'oras://community.wave.seqera.io/library/hmmer_prodigal-gv_biopython_pandas:de876e9113694da7' :
        'community.wave.seqera.io/library/hmmer_prodigal-gv_biopython_pandas:aaaf7b9a9207df90' }"

    input:
    tuple val(meta) , path(fasta)
    tuple val(meta2), path(faa)

    output:
    tuple val(meta), path("${prefix}.lengthfilter.fasta.gz")    , emit: fasta
    tuple val(meta), path("${prefix}.lengthfilter.faa.gz")      , emit: faa
    path "versions.yml"                                         , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    fastafaalengthfilter.py \\
        --input_fasta ${fasta} \\
        --input_faa ${faa} \\
        --prefix ${prefix}.lengthfilter \\
        ${args}

    gzip ${prefix}.lengthfilter.fasta ${prefix}.lengthfilter.faa

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
        biopython: \$(python -c "import Bio; print(Bio.__version__)")
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    echo "" | gzip > ${prefix}.lengthfilter.fasta.gz
    echo "" | gzip > ${prefix}.lengthfilter.faa.gz

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
        biopython: \$(python -c "import Bio; print(Bio.__version__)")
    END_VERSIONS
    """
}
