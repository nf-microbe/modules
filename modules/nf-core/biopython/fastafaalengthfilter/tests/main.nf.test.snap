{
    "fasta.gz + faa.gz": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.lengthfilter.fasta.gz:md5,238c9c4b8fd2a3f067bf3e840ead459a"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.lengthfilter.faa.gz:md5,fbcb4274ae897b730beff303fbf2c20c"
                    ]
                ],
                "2": [
                    "versions.yml:md5,7dc973a028195c796084df7fbc23d31d"
                ],
                "faa": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.lengthfilter.faa.gz:md5,fbcb4274ae897b730beff303fbf2c20c"
                    ]
                ],
                "fasta": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.lengthfilter.fasta.gz:md5,238c9c4b8fd2a3f067bf3e840ead459a"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,7dc973a028195c796084df7fbc23d31d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-29T15:22:30.152540768"
    },
    "stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.lengthfilter.fasta.gz:md5,238c9c4b8fd2a3f067bf3e840ead459a"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.lengthfilter.faa.gz:md5,fbcb4274ae897b730beff303fbf2c20c"
                    ]
                ],
                "2": [
                    "versions.yml:md5,7dc973a028195c796084df7fbc23d31d"
                ],
                "faa": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.lengthfilter.faa.gz:md5,fbcb4274ae897b730beff303fbf2c20c"
                    ]
                ],
                "fasta": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.lengthfilter.fasta.gz:md5,238c9c4b8fd2a3f067bf3e840ead459a"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,7dc973a028195c796084df7fbc23d31d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-29T15:22:42.341823727"
    }
}